source/inmem/memory.go

```diff
@@ -0,0 +1,189 @@
+package inmem
```

Comment:
For built-in CLI support, you'll also need to add a `build_mem.go` file in `internal/cli`. See other files there for examples.

Reply:
There's a bit of overlap with the existing `stub` driver. Thoughts about using that or improving it instead? For usage, see `source/stub/stub_test.go`.

Reply:
when I first coding this, I thought the same way, but thinks that importing from another package is okay in this project. But, since `stub` is likely just for testing purpose and have simple logic, I will use that